src/librustc_typeck/diagnostics.rs

```diff
@@ -1490,21 +1490,49 @@ For information on the design of the orphan rules, see [RFC 1023].
 
 E0118: r##"
 Rust can't find a base type for an implementation you are providing, or the type
-cannot have an implementation. For example, only a named type or a trait can
-have an implementation:
+cannot have an implementation. Erroneous code examples:
```

Comment:
The whole "base type" thing isn't quite right: "can't find" really means "there isn't".
And "base type" refers to an user-defined type.
So what it's trying to say is that the type is not an user-defined type that can have an impl, AFAIK.